File: SDE_sheet/Arrays/06_stock_buy_and_sell/solution.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if len(prices) == 0:
            return 0
        maxProfit = 0
        for i in range(1, len(prices)):
            if prices[i] < prices[0]:
                prices[0] = prices[i]
            else:
                maxProfit = max(maxProfit, prices[i] - prices[0])
        return maxProfit

File: SDE_sheet/Arrays/06_stock_buy_and_sell/solution.java
class Solution {
    public int maxProfit(int[] prices) {
        if (prices.length == 0) return 0;
        int maxProfit = 0;
        for (int i = 1; i < prices.length; i++) {
            if (prices[i] < prices[0]) {
                prices[0] = prices[i];
            } else {
                maxProfit = Math.max(maxProfit, prices[i] - prices[0]);
            }
        }
        return maxProfit;
    }
}
